ci.py

```diff
@@ -0,0 +1,200 @@
+                    self.global_branch_name = conf['branch']
+                    self.set_global_branches()
+                    break
+
+            # Yep we have high complexity here but list is short
+            for conf in self.custom_config['deps']:
+                if conf['name'] == 'global':
+                    continue
+                for dep in self.config['deps']:
+                    if dep['name'] == conf['name']:
+                        # TODO add support for other fields
+                        dep['branch'] = conf['branch']
+                        break
+                else:
+                    logging.error(f'Can"t find {conf["name"]} repo. We have list of repos in ci-default.yml')
+                    sys.exit(1)
+
+    def set_global_branches(self):
+        for dep in self.config['deps']:
+            url = dep['url']
+            g = git.cmd.Git()
+            # TODO maybe it'll be faster to use local data
+            output = g.ls_remote("--heads", url, self.global_branch_name)
+            if self.global_branch_name in output:
+                logging.info(f'Use branch {self.global_branch_name} for {dep["name"]}')
+                dep['branch'] = self.global_branch_name
+
+    def create_fb(self, branch_name):
+        import git
+        repo = git.Repo('.')
+
+        git = repo.git
+        for ref in repo.references:
+            if branch_name == ref.name:
+                git.checkout(branch_name)
+                break
+        else:
+            git.checkout('HEAD', b=branch_name)
+
+        if self.custom_config is not None:
+            for dep in self.custom_config['deps']:
+                if dep['name'] == 'global':
+                    dep['branch'] = branch_name
+        else:
+            global_branch = {'name': 'global', 'branch': branch_name}
+            self.custom_config = {'deps': [global_branch,]}
+
+        self.write_custom_config(self.custom_config)
+        repo.git.add(all=True)
+        repo.index.commit('Create fuature build')
+        origin = repo.remote(name='origin')
+        logging.info('Branch was created')
+        logging.info(f'Need to create PR now: https://github.com/Percona-Lab/pmm-submodules/compare/{branch_name}?expand=1')
+
+    def get_deps(self, single_branch=False):
+        with open(GIT_SOURCES_FILE, 'w+') as f:
+            f.truncate()
+
+        with open(GIT_SOURCES_FILE, 'a') as f:
+            for dep in self.deps:
+                path = os.path.join(self.rootdir, dep["path"])
+                if not os.path.exists(os.path.join(self.rootdir, path)):
+                    target_branch = dep['branch']
+                    target_url = dep["url"]
+                    if single_branch:
+                        check_call(f'git clone --depth 1 --single-branch --branch {target_branch} {target_url} {path}'.split())
+                    else:
+                        check_call(f'git clone --depth 1 --no-single-branch {target_url} {path}')
```

Comment:
what case we might want to clone all branches in?

Reply:
I was just comparing the difference in speed.